### PR TITLE
added REST API endpoint GET /_admin/server/availability

### PR DESCRIPTION
**Pull request by jsteemann**
This a public API, which can be used without authentication.
It is meant to be used in the context of server monitoring only

### CODE REVIEW

**Comment by graetzer** [comment]
I think you also need to add a check to `GeneralCommTask::canAccessPath`

**Comment by jsteemann** [comment]
@graetzer : ok, will check!

File: Documentation/DocuBlocks/Rest/Administration/get_admin_server_availability.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@RESTRETURNCODES
+
+@RESTRETURNCODE{200}
+This API will return HTTP 200 in case the server is up and running and usable for
+arbitrary operations, is not set to read-only mode and is currently not a follower 
+in case of an active failover setup.
+
+@RESTRETURNCODE{503}
+HTTP 503 will be returned in case the server is during startup or during shutdown,
+is set to read-only mode or is currently not a follower in an active failover setup.
```
Comment (sleto-it) [inline Documentation/DocuBlocks/Rest/Administration/get_admin_server_availability.md]: "is currently not a follower" --> remove not

File: Documentation/DocuBlocks/Rest/Administration/get_admin_server_availability.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+@startDocuBlock get_admin_server_availability
+@brief Get to know whether this server is a Coordinator or DB-Server
```
Comment (sleto-it) [inline Documentation/DocuBlocks/Rest/Administration/get_admin_server_availability.md]: "Get to know whether this server is a Coordinator or DB-Server" --> copy / paste error? probably it is "Return whether or not a server is available"

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/390/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/394/

**Pull request merged**